### DISCUSSION TITLE
Accessing the previous NL values

### DISCUSSION THREAD

**Discussion by doquang** [idaholab/moose#26503] [Q&A General] [answered]
Hi,

I'm trying to writte a new BC class. In this class. I would like to assign values from the `MooseVariableFE::dofValuesPreviousNL()` function to my `VariableValue`, but I got an error when executing this code:
```
The following error occurred in the object "MOOSE Problem", of type "FEProblem".
Vector tag 'U_PREVIOUS_NL_NEWTON' does not exist
```

Could you please tell me how to access the previous Newton iterate of a coupled variable.

Thanks in advance,
/ Minh

**Comment by GiudGiud** [comment, marked as answer]
hello

To use this routine, you need to add this option to the `[Problem]` block: previous_nl_solution_required 

Guillaume
